File: practice/string/doubleChar.java
/* Greg Murashige


Given a string, return a string where for every char in the
original, there are two chars.


doubleChar("The") → "TThhee"
doubleChar("AAbb") → "AAAAbbbb"
doubleChar("Hi-There") → "HHii--TThheerree"
*/
public String doubleChar(String str) {
  if (str.isEmpty()) return str;

  String lastChar = String.valueOf(str.charAt(str.length()-1));
  lastChar += lastChar; //double the last char

  return (doubleChar(str.substring(0, str.length()-1)) +  lastChar);
}

File: practice/string/double_char.py
""" Greg Murashige
Recursion because.


Given a string, return a string where for every char in the
original, there are two chars.


double_char('The') → 'TThhee'
double_char('AAbb') → 'AAAAbbbb'
double_char('Hi-There') → 'HHii--TThheerree'
"""

def double_char(str):
  if len(str) == 0:
    return str
  return str[0] + str[0] + double_char(str[1:])
